code/modules/spells/targeted/street_alchemy.dm

```diff
@@ -0,0 +1,177 @@
+	var/obj/item/weapon/reagent_containers/pill/streetAlchElixir/elixir = null
+	if(spell_levels[Sp_POWER])
+		elixir = new /obj/item/weapon/reagent_containers/pill/streetAlchElixir/hypoElixir(src)
+	else
+		elixir = new /obj/item/weapon/reagent_containers/pill/streetAlchElixir(src)
+	if(user.find_empty_hand_index())
+		user.put_in_hands(elixir)
+	if(iscarbon(target))
+		mobAlchemy(target, user, elixir)
+	if((istype(target, /obj/structure)) || (istype(target, /obj/item/weapon/storage)) || (istype(target, /obj/item/weapon/reagent_containers)) || (istype(target, /obj/machinery)))
+		itemAlchemy(target, user, elixir)
+	if(elixir.is_empty())
+		to_chat(user, "You fail to perform alchemy.")
+		user.drop_item(elixir, force_drop = 1)
+		qdel(elixir)
+		return
+	existingElixirs += elixir
+	if(existingElixirs.len > elixirAmount)
+		var/tooMany = existingElixirs[1]
+		existingElixirs -= tooMany
+		qdel(tooMany)
+	playsound(user, "sound/effects/bubbles.ogg", 75, 1)
+
+/spell/targeted/alchemy/proc/aoeAlchemy(target, mob/user)
+	var/obj/item/weapon/reagent_containers/pill/streetAlchElixir/elixir = null
+	if(spell_levels[Sp_POWER])
+		elixir = new /obj/item/weapon/reagent_containers/pill/streetAlchElixir/hypoElixir(src)
+	else
+		elixir = new /obj/item/weapon/reagent_containers/pill/streetAlchElixir(src)
+	if(user.find_empty_hand_index())
+		user.put_in_hands(elixir)
+	for(var/T in range(1, target))
+		if(iscarbon(T))
+			mobAlchemy(T, user, elixir)
+		if((istype(T, /obj/structure)) || (istype(T, /obj/item/weapon)) || (istype(T, /obj/machinery)))
+			itemAlchemy(T, user, elixir)
+	if(elixir.is_empty())
+		to_chat(user, "You fail to perform alchemy.")
+		user.drop_item(elixir, force_drop = 1)
+		qdel(elixir)
+		return
+	existingElixirs += elixir
+	if(existingElixirs.len > elixirAmount)
+		var/tooMany = existingElixirs[1]
+		existingElixirs -= tooMany
+		qdel(tooMany)
+	playsound(user, "sound/effects/bubbles.ogg", 75, 1)
+
+/spell/targeted/alchemy/proc/mobAlchemy(var/mob/living/carbon/C, mob/user, var/obj/item/weapon/reagent_containers/pill/elixir)
+	var/numThefts = 0
+	C.reagents.trans_to(elixir, C.reagents.total_volume)
+	numThefts++
+		for(var/S in get_contents_in_object(C))
+			if((numThefts >= reagToSteal) || (elixir.reagents.is_full()))
+				break
+			var/obj/R = S
+			if((R.reagents) && (R.reagents.total_volume > 0))
+				numThefts++
+				R.reagents.trans_to(elixir, R.reagents.total_volume)
+	playsound(C, "sound/effects/bubbles.ogg", 75, 1)
+
+/spell/targeted/alchemy/proc/itemAlchemy(var/obj/C, mob/user, var/obj/item/weapon/reagent_containers/pill/elixir)
+	var/numThefts = 0
+	if(C.reagents)
+		C.reagents.trans_to(elixir,C.reagents.total_volume)
+		numThefts++
+	if(C.contents)
+		for(var/S in get_contents_in_object(C))
+			if((numThefts >= reagToSteal) || (elixir.reagents.is_full()))
+				break
+			var/obj/R = S
+			if((R.reagents) && (R.reagents.total_volume > 0))
+				numThefts++
+				R.reagents.trans_to(elixir, R.reagents.total_volume)
```

Comment:
It seems like these two could be replaced by a proc that sucks reagents out of an atom type object, since atom is a common ancestor of both mob and obj and it has reagents.

This might introduce some unforeseen consequences though.

Also it's weird that only one of these two makes sound, though I imagine you might have done that to prevent recursive bubbling sounds.

Speaking of recursive, how safe are beakers inside boxes inside backpacks?

Reply:
Or they could be used as wrappers for a common proc that treats both as atom

Reply:
I was planning on it but I realized I couldn't/shouldn't for some reason. I cannot remember that reason. I'll review it to see if I was wrong/remember the reason in a bit. 

>how safe are beakers inside boxes inside backpacks?

Not safe
Originally I had it only search a layer deep but then I realized get_contents_in_object() isn't mob specific, and that searches everything. 
>one of these two makes sound

I barely understand how playsound works and kind of thought only the target could hear it, which makes no sense now that I think about it. "Why would a beaker need to hear a sound". You're right I'll add that after that review.

Reply:
playsound will play a sound at the turf in question, there is a way to play a sound for just one person, but most commonly we want sounds to be heard by anyone who isn't deaf, etc. - it also handles things like atmosphere

Reply:
Made the change, I think the reason I thought I couldn't was not knowing atoms had reagents. Added the sound too.